### DISCUSSION TITLE
[Release Vote] Vote for Release V0.4.1-rc0

### DISCUSSION THREAD

**Discussion by baiyangtx** [apache/amoro#1226] [Polls]
Hi Everyone,

I propose that we release the following RC as the official Arctic 0.4.1 release.

The commit ID is https://github.com/NetEase/arctic/commit/e7993c1b3adedbbadae4570e73a0843bd8787528

This corresponds to the tag: v0.4.1-rc0
https://github.com/NetEase/arctic/commits/v0.4.1-rc0
https://github.com/NetEase/arctic/tree/v0.4.1-rc0
Please check out, build, and test.

[ ] +1 Release this as Arctic 0.4.1
[ ] +0
[ ] -1 Do not release this because...

Please vote in the next 72 hours and add your test steps when voting.

**Comment by zhoujinsong** [comment]
+1

For testing the release in my local environment with the following step:

- Ran build and test with JDK8 and JDK11
- Ran the quick start guild
- Created the iceberg format catalog and table

**Comment by wangtaohz** [comment]
+0

I tested it locally :
- Ran the unit test of core/ams/hive/spark/optimizer locally
- Ran optimizing integration test locally
- Installed the local ams environment, and ran the quick start guild
- Tested Optimizing of Iceberg format table and Mixed Iceberg format table

But I think another two things have to be done before release 0.4.1
- change the version in pom.xml to 0.4.1
- merge PR https://github.com/NetEase/arctic/pull/1229 to fix `Trash` for UnkeyedTable

**Reply by wangtaohz** [reply]
The version for v0.4.1-rc0(e51cc514d6218798a6) is 0.4.1 now.

**Comment by lklhdu** [comment]
+0

I tested it locally :

* Ran the unit test of core/ams/hive/spark/flink/optimizer locally
* Tested read and write functionality of hive tables, covering logstore and filestore

But I think one thing has to be done before release 0.4.1

* merge PR https://github.com/NetEase/arctic/pull/1223 to fix NotSerializableException for Flink Sink Function

**Comment by YesOrNo828** [comment]
-1 
- Because the project version is 0.5.0-snapshot, it should be 0.4.1 
- Create catalog `demo_catalog` with `warehouse` hdfs://namenode:8020/user/arctic/demo_warehouse in the quickstart guide. I should suggest user change the actual `namenode` name in the guide. PR: https://github.com/NetEase/arctic/pull/1250

What I did:
- Build the project with all modules including the Trino module successfully.
- Ran the quickstart with the binary package. create a keyed table, insert, and select the table with spark and flink engines.
